src/spikeinterface/core/numpyextractors.py

```diff
@@ -347,13 +347,16 @@ def get_unit_spike_train(self, unit_id, start_frame, end_frame):
             s0, s1 = np.searchsorted(self.spikes["segment_index"], [self.segment_index, self.segment_index + 1])
             self.spikes_in_seg = self.spikes[s0:s1]
 
+        start = 0 if start_frame is None else np.searchsorted(self.spikes_in_seg["sample_index"], start_frame)
+        end = (
+            len(self.spikes_in_seg)
+            if end_frame is None
+            else np.searchsorted(self.spikes_in_seg["sample_index"], end_frame)
+        )
+
         unit_index = self.unit_ids.index(unit_id)
-        times = self.spikes_in_seg[self.spikes_in_seg["unit_index"] == unit_index]["sample_index"]
+        times = self.spikes_in_seg[start:end][self.spikes_in_seg[start:end]["unit_index"] == unit_index]["sample_index"]
```

Comment:
This will be easier if we did not use the concatenated version of `to_spike_vector`

Reply:
I do not understand?

Reply:
Unrelated to this PR. But the `to_spike_vector` function has a function to return a list of spike_vectors per segment instead of all the spike_vectors concanteted, so you could avoid the first np.searchsorted which look for the segment indexes.

Reply:
https://github.com/SpikeInterface/spikeinterface/blob/329197618a9b48ef876d1e8b8e79f07f4abf5e49/src/spikeinterface/core/basesorting.py#L427

`concatenated` keyword.

Reply:
Ah ok!

I didn't touch that, but I removed the disgusting `times[times > start]` which doesn't take advantage of the fact that it is ordered!

Reply:
"disgusting" is a bit hard but I get the feeling. Note that generally this `start_frame` \  `end_frame` is very often None.

Reply:
@h-mayorquin 
Now it use to_spike_vector(concatenated=False) which is better.